internal/db/fetcher/fetcher.go

```diff
@@ -278,11 +278,20 @@ func (df *DocumentFetcher) start(ctx context.Context, spans []core.Span, withDel
 	} else {
 		valueSpans := make([]core.Span, len(spans))
 		for i, span := range spans {
-			// We can only handle value keys, so here we ensure we only read value keys
 			if withDeleted {
-				valueSpans[i] = core.NewSpan(span.Start.WithDeletedFlag(), span.End.WithDeletedFlag())
+				// DocumentFetcher only ever recieves document keys
+				//nolint:forcetypeassert
+				valueSpans[i] = core.NewSpan(
+					span.Start.(keys.DataStoreKey).WithDeletedFlag(),
+					span.End.(keys.DataStoreKey).WithDeletedFlag(),
+				)
 			} else {
-				valueSpans[i] = core.NewSpan(span.Start.WithValueFlag(), span.End.WithValueFlag())
+				// DocumentFetcher only ever recieves document keys
+				//nolint:forcetypeassert
```

Comment:
suggestion: Have a must DataStoreKey function for the casting? and do this lint ignore there once for all these 4 casts?

Reply:
I do not think moving a simple cast into a function would improve the readability of the code here.

Reply:
Can also move `span.Start.(keys.DataStoreKey)` and `span.End.(keys.DataStoreKey)` out of if-else blocks into a variable, and then use that in their respective occurrences. Feel free to ignore as this seems like subjective pref